lib/functions.php

```diff
@@ -191,11 +191,18 @@ function html_email_handler_send_email(array $options = null) {
 
 		// convert to to correct format
 		$to = implode(", ", $options["to"]);
-		
+
 		// encode subject to handle special chars
 		$subject = "=?UTF-8?B?" . base64_encode($options["subject"]) . "?=";
-			
-		$result = mail($to, $subject, $message, $headers, $sendmail_options);
+
+		global $CONFIG;
+		if (!isset($CONFIG->block_mail)) {
```

Comment:
block_mail in CONFIG is probably something client specific as it does not exists in Elgg by default. We can't add support for that. And by the way there is a better way to prevent these mails, by unregistering the hook and register your own email_handler